README.md

```diff
@@ -14,27 +39,804 @@ gem 'uploadcare-rails'
+```
+
+#### Uploadcare File Group
+
+Groups work similar to the File but have some differences though.
+
+```ruby
+# app/models/post.rb
+class Post < ApplicationRecord
+  mount_uploadcare_file_group :attachments
+end
+```
+
+Creating a new `post` with the Group mounted:
+
+```console
+$ post = Post.create(attachments: 'https://ucarecdn.com/dbc4e868-b7a6-43ff-a35f-2ebef935dc1b~1/')
+```
+
+Now the `post.attachments` is an Uploadcare::Rails::Group. Following methods are supported:
+
+```console
+# Store the file group on an Uploadcare server permanently:
+$ post.attachments.store
+#   => {
+#         "cdn_url"=>"https://ucarecdn.com/dbc4e868-b7a6-43ff-a35f-2ebef935dc1b~1/",
+#          ...other group data...
+#         "files"=> [{
+#            "datetime_stored"=>"2021-07-29T08:31:45.668354Z",
+#            ...other file data...
+#         }]
+#      }
+#   
+# Delete the file group from an Uploadcare server permanently:
+$ post.attachments.delete
+#   => {
+#         "datetime_removed"=>"2021-07-30T09:19:30.797174Z",
+#          ...other group data...
+#      }
+#
+# Get CDN-url of an object attribute:
+$ post.attachments.to_s
+#   => "https://ucarecdn.com/dbc4e868-b7a6-43ff-a35f-2ebef935dc1b~1/"
+#
+# Load object - works the same way as for the File:
+$ post.attachments.load
+#   => {
+#         "cdn_url"=>"https://ucarecdn.com/dbc4e868-b7a6-43ff-a35f-2ebef935dc1b~1/",
+#          ...other group data...
+#         "files"=> [{
+#            "datetime_stored"=>"2021-07-29T08:31:45.668354Z",
+#            ...other file data...
+#         }]
+#      }
+#
+# Check if an attribute loaded from the server:
+$ post.attachments.loaded?
+#   => true
+#
+# As we don't want to show (on the html-page) a file group itself,
+# we can get CDN-urls for file that the group contains. No loading group or files needed.
+# This works for images only:
+$ post.attachments.transform_file_urls(quality: 'better')
+#   => ["https://ucarecdn.com/dbc4e868-b7a6-43ff-a35f-2ebef935dc1b~1/nth/0/-/quality/better/"]
+#
+# If you want to get non-transformed file urls, use:
+$ post.attachments.file_urls
+#   => ["https://ucarecdn.com/dbc4e868-b7a6-43ff-a35f-2ebef935dc1b~1/nth/0/"]
+```
+
+
+### Image Transformation
+
+Uploadcare provides a way to transform images stored on Uploadcare services specifying a list of operations.
+If an operation has just one option, you can specify it like key-value:
+
+```console
+$ post.picture.transform_url(quality: 'better')
+#   => "https://ucarecdn.com/ebbb9929-eb92-4f52-a212-eecfdb19d27d/-/quality/better/"
+```
+
+and if an operation supports several options - just set them as a Hash:
+
+```console
+$ post.picture.transform_url(crop: { dimensions: '300x500', coords: '50, 50', alignment: 'center' })
+#   => "https://ucarecdn.com/ebbb9929-eb92-4f52-a212-eecfdb19d27d/-/crop/300x500/50,50/center/"
+```
+
+Full list of operations and valid values can be found [here](https://uploadcare.com/docs/transformations/image/#image-transformations).
```

Comment:
No for now. It works like a filter, i.e. it builds an image URL with only "allowed" operations. If an operation is not known by the gem or known operations has no value - it will be omitted. But we can build an URL with all the operations specified by users if you want.

Reply:
We add new operations regularly, so we need something like that.

Reply:
Ok, got it

Reply:
Done